### PR TITLE
revamp `eslint` config

### PR DESCRIPTION
**Pull request by with-heart**
This draft PR is the start of work towards revamping our `eslint` config so it can better support all the different things we're working on in this here monorepo. 

Opening as a draft PR because these changes cause a decent number of new errors throughout the codebase, so we need to go through and resolve them in one of two ways:

- change to `warn` or `off`
- fix the code

Other things:

- make sure that the `next` stuff is configured correctly for the `app` directories (minus storybook). Their configuration is really great so definitely want to keep that
- remove `eslint-disable` directives (216 lines in various files) and resolve them as well

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/** @type {import('eslint').ESLint.ConfigData} */
+const config = {
+  root: true,
+  env: {
+    node: true,
+  },
+  parser: "@typescript-eslint/parser",
```
Comment (with-heart) [inline .eslintrc.cjs:7]: Switching to [`@typescript-eslint/parser`](https://typescript-eslint.io/) adds TypeScript support to `eslint`

File: .eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  plugins: ["@typescript-eslint", "simple-import-sort"],
+  extends: ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
```
Comment (with-heart) [inline .eslintrc.cjs:9]: I'm not entirely sure this is correct. It's possible that we should only enable the `@typescript-eslint`-related options in the `**/*.ts`/`**/*.tsx` block below, but I need to play around to figure that out

Comment (with-heart) [inline .eslintrc.cjs:9]: `eslint:recommended` enables the recommended base set of rules and `plugin:@typescript-eslint/recommended` enables its recommended base set of rules which enable it to support TypeScript syntax.

`eslint:recommended` has some rules that conflict with the TypeScript rules so `@typescript-eslint/recommended` disables them

File: .eslintrc.cjs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  overrides: [
```
Comment (with-heart) [inline .eslintrc.cjs:10]: Overrides allow us to specify `eslint` configuration blocks that target the files listed under the `files` key

File: .eslintrc.cjs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      files: [
+        "*.js",
+        "*.cjs",
+        "*.mjs",
+        "apps/*/*.js",
+        "apps/*/*.cjs",
+        "apps/*/*.mjs",
+        "packages/*/*.js",
+        "packages/*/*.cjs",
+        "packages/*/*.mjs",
```
Comment (with-heart) [inline .eslintrc.cjs:21]: We're going to have a decent amount of `js`-based configuration files scattered around (just the nature of a monorepo) so we want to make sure we lint them too

File: .eslintrc.cjs [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        tsconfigRootDir: __dirname,
+        project: [
+          "./tsconfig.eslint.json",
+          "./apps/*/tsconfig.json",
+          "./packages/*/tsconfig.json",
+        ],
+        extends: [
+          "plugin:@typescript-eslint/recommended-requiring-type-checking",
+        ],
+      },
```
Comment (with-heart) [inline .eslintrc.cjs:39]: This block adds type-aware linting powered by `typescript-eslint`. Note that `project` defines the root first (`./tsconfig.eslint.json`) and package directories after.

File: .eslintrc.cjs [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      env: {
+        browser: true,
+      },
+      settings: {
+        next: {
+          rootDir: ["apps/*"],
+        },
+      },
+      extends: ["plugin:@next/next/recommended", "next"],
+    },
```
Comment (with-heart) [inline .eslintrc.cjs:52]: This block is for NextJS app-specific stuff

Comment (with-heart) [reply .eslintrc.cjs:52]: I should add these same notes as comments in the file

File: tsconfig.eslint.json [hunk 7]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "@tsconfig/recommended/tsconfig.json",
+  "compilerOptions": {
+    "noEmit": true
+  },
+  "include": ["**/*.ts", "**/*.tsx"]
+}
```
Comment (with-heart) [inline tsconfig.eslint.json:7]: This is a basic `tsconfig` file that's only used by `eslint`. It allows us to add a wide glob `**/*` covering all `ts`/`tsx` files that might not fall under the package-specific `tsconfig` files.

**Comment by e-roy** [comment]
> 

@with-heart this will be easier to do after switching to the new api and consolidating apps back down to 2 plus storybook.  a lot of these `eslint-disable` directives will resolve automatically 

what is the best way to find way to spot the `eslint-disable` directives used without individual opening every file?   

common errors: 
camelcase errors from type naming from the db
changing useEffect to useCallback in some cases
callback functions to parent needing to resolve val better

**Pull request closed without merging**